ClinGen/clincoded
# Issue #1746: User Registration

**Steps to test #1746**

**Note:** Confluence doc can be found in the JIRA issue.

1. On the home page in the center grey box, check that the links work and the text makes sense.
2. Click the "Login/Sign Up" button in the top right.
3. Click the "Sign Up" tab and fill in the information with a real email that you have access to and tick the box for the user agreement. (Alternatively, you can sign up with Google Sign Up. If you choose this method, you will skip steps 4, 5, and 6.)
4. Expect to be routed to a page the explains the next steps in the sign up process. Read steps and make sure it makes sense.
5. Login to your email and look for an email verification email from Auth0. Click the button in the email to verify your account.
6. Go back to the curation interfaces and login with your new credentials.
7. You will be directed to a "Thank you for registering" page. Check that the text makes sense.
8. Login to your existing account (that has admin access)
9. Go to /users/ and find your new user at the top of the table with a user status of "Requested Activation"
10. Change your user to "Active" in the dropdown and click "Save" at the bottom of the page.
11. Log out of your admin account and log in to your freshly activated user.
12. Click around the interfaces to ensure you have appropriate access to the VCI and GCI.

-----
**Steps to test user upgrade step for user's date added**
1. Go to /src/clincoded/schemas/user.json and change the schema version to 5.
2. Start the backend
3. Go to the same schema file and change the schema version to 6.
4. Open a new terminal and run "bin/batchupgrade development.ini --app-name app"
5. Run the frontend
6. Login as an admin and go to /users/. Check to make sure none of the users have a "Dated Added" from the year 1994. Make sure all users have a creation time of 12:00am.
